genesyscloud/external_contacts_organization/resource_genesyscloud_externalcontacts_organization_utils.go

```diff
@@ -0,0 +1,407 @@
+package external_contacts_organization
+
+import (
+	"encoding/json"
+	"fmt"
+	"terraform-provider-genesyscloud/genesyscloud/util"
+	"terraform-provider-genesyscloud/genesyscloud/util/lists"
+	"terraform-provider-genesyscloud/genesyscloud/util/resourcedata"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mypurecloud/platform-client-sdk-go/v146/platformclientv2"
+	"github.com/nyaruka/phonenumbers"
+)
+
+/*
+The resource_genesyscloud_external_contacts_organization_utils.go file contains various helper methods to marshal
+and unmarshal data into formats consumable by Terraform and/or Genesys Cloud.
+*/
+
+// getExternalContactsOrganizationFromResourceData maps data from schema ResourceData object to a platformclientv2.Externalorganization
+func getExternalContactsOrganizationFromResourceData(d *schema.ResourceData) (platformclientv2.Externalorganization, error) {
+	externalOrganization := platformclientv2.Externalorganization{
+		Name:                platformclientv2.String(d.Get("name").(string)),
+		CompanyType:         platformclientv2.String(d.Get("company_type").(string)),
+		Industry:            platformclientv2.String(d.Get("industry").(string)),
+		Address:             buildSdkAddress(d, "address"),
+		PhoneNumber:         buildSdkPhoneNumber(d, "phone_number"),
+		FaxNumber:           buildSdkPhoneNumber(d, "fax_number"),
+		EmployeeCount:       platformclientv2.Int(d.Get("employee_count").(int)),
+		Revenue:             platformclientv2.Int(d.Get("revenue").(int)),
+		Tickers:             buildTickers(d.Get("tickers").([]interface{})),
+		TwitterId:           buildSdkTwitterId(d, "twitter"),
+		ExternalSystemUrl:   platformclientv2.String(d.Get("external_system_url").(string)),
+		Trustor:             buildTrustor(d.Get("trustor").([]interface{})),
+		ExternalDataSources: buildExternalDataSources(d.Get("external_data_sources").([]interface{})),
+	}
+	tags := lists.InterfaceListToStrings(d.Get("tags").([]interface{}))
+	websites := lists.InterfaceListToStrings(d.Get("websites").([]interface{}))
+	externalOrganization.Tags = &tags
+	externalOrganization.Websites = &websites
+	if d.Get("primary_contact_id").(string) != "" {
+		externalOrganization.PrimaryContactId = platformclientv2.String(d.Get("primary_contact_id").(string))
+	}
+	schema, err := BuildOrganizationSchema(d.Get("schema").([]interface{}))
+	if err != nil {
+		return externalOrganization, err
+	}
+	externalOrganization.Schema = schema
+	customFields, err := buildCustomFieldsNillable(d.Get("custom_fields").(string))
+	if err != nil {
+		return externalOrganization, err
+
+	}
+	externalOrganization.CustomFields = customFields
+
+	return externalOrganization, nil
+}
+
+// buildPhonenumberFromData is a helper method to map phone data to the GenesysCloud platformclientv2.PhoneNumber
+func buildPhonenumberFromData(phoneData []interface{}) *platformclientv2.Phonenumber {
+	phoneMap := phoneData[0].(map[string]interface{})
```

Comment:
A type-check here could help avoid a panic error in the future 

```go
phoneMap, ok := ...
if !ok {
  return nil
}
```

Reply:
It looks like the calling function does a nil check, do we still need to include it?
<img width="666" alt="image" src="https://github.com/user-attachments/assets/82b3b38f-7bba-477b-9a23-7e660a14d587">

Reply:
I think that would be checking that the interface array is not nil. Here we're casting the item inside the array. So if phoneData has one nil item, phoneData is not nil but we will see a panic error if we try to cast the item inside phoneData to a map without the `, ok :=`

Reply:
This can happen if the user defines phone_number and doesn't specify any attributes 

```
phone_number {}
```